docs/cip/README.md

```diff
@@ -4,704 +4,2800 @@ Title: Ouroboros Leios - Greater transaction throughput
+that nodes already possess all necessary EBs from alternative forks, eliminating
+additional propagation delays during fork switches.
+
+<a id="mempool-capacity" href="#mempool-capacity"></a>**Mempool Capacity
+Requirements**: The mempool must accommodate both RB and EB transaction
+production. The capacity requirements are significantly increased compared to
+Praos to handle the additional transaction volume expected from EB production.
+
+#### Epoch Boundary
+
+<a id="persistent-voter-computation" href="#persistent-voter-computation"></a>**Persistent
+Voter Computation**: Nodes must compute the set of persistent voters for each
+epoch using the [Fait Accompli scheme][fait-accompli-sortition]. This
+computation uses the stake distribution that becomes available at the epoch
+boundary and represents a minimal computational overhead based on current
+[BLS certificates benchmarks](https://github.com/input-output-hk/ouroboros-leios/blob/main/crypto-benchmarks.rs/Specification.md#benchmarks-in-rust).
+Nodes should complete this computation well before voting begins in the new
+epoch to ensure seamless participation.
+
+### Network
+
+As outlined above, Leios splits transactions between RBs and EBs, with EB
+inclusion dependent on committee voting and certification. **Transactions in
+RBs**: EB certificate generation requires an absolute stake majority. Given that
+this may not always be the case, if transactions were only included in EBs, the
+protocol would risk **losing liveness** due to no EB certificate being
+generated. As a remedy, RBs are allowed to directly contain transactions,
+ensuring basic liveness even when EB certification fails.
+
+Unlike Ouroboros Praos where the RB chain contains all necessary data, Leios
+nodes require additional message types to:
+
+- **Reconstruct ledger state**: EBs containing certified transactions
+- **Participate in consensus**: Vote on EBs and construct certificates
+- **Detect equivocation**: RB headers from competing forks
+
+#### Praos Mini-Protocols
+
+As described in [Node Behavior](#node-behavior), existing Praos mini-protocols
+continue to operate with only minor modifications to support Leios. ChainSync
+exchanges RB headers that now include optional fields for EB announcements
+(`announced_eb`) and certifications (`certified_eb`). BlockFetch retrieves RB
+bodies that may contain BLS aggregate certificates (`eb_certificate`) alongside
+standard transactions. TxSubmission remains unchanged except for expanded
+mempool capacity to support both RB and EB transaction pools.
+
+#### Leios Mini-Protocols
+
+Leios will introduce new mini-protocols to handle the additional message types
+required for EB distribution, voting, and certificate construction.
+
+This CIP does not specify how exactly Leios nodes exchange messages. Those
+details must eventually be specified so that different node implementations can
+interoperate, but that standardization can be deferred to a follow-up CIP. The
```

Comment:
Could the details be specified in a Cardano Blueprint document? or do we really need a follow-up CIP?

Reply:
I already suggested that we should be more confident with what we propose here and that the CIP can also be updated: https://github.com/input-output-hk/ouroboros-leios/pull/513#pullrequestreview-3146008586

Ensuring implementation-independent specification is available on the `cardano-blueprint` is part of the acceptance criteria + path to active.